stacks/ring.go

```diff
@@ -14,7 +14,11 @@ type ring struct {
 	end int
 }
 
+//writes b[] into the ring buffer at the current write position (r.end)  - wrapping if needed, and increasing r.end (possibly wrapping)
```

Comment:
Always start function docstrings with the function name, preceded by a space, so: `// Write ...`